cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_best_split_sp_opt_impl_dpc.cpp

```diff
@@ -0,0 +1,1143 @@
+
+                                reduce_hist_over_group(item,
+                                                       local_buf_float_ptr,
+                                                       left_count,
+                                                       mean,
+                                                       sum2cent);
+
+                                ts.left_count = Index(left_count);
+                                ts.right_count = row_count - left_count;
+
+                                ts.left_hist[0] = left_count;
+                                ts.left_hist[1] = mean;
+                                ts.left_hist[2] = sum2cent;
+                            }
+
+                            if (local_id == 0) {
+                                if constexpr (std::is_same_v<Task, task::classification>) {
+                                    sp_hlp.calc_imp_dec(ts,
+                                                        node_ptr,
+                                                        node_imp_list_ptr,
+                                                        class_hist_list_ptr,
+                                                        class_count,
+                                                        node_id);
+                                    sp_hlp.choose_best_split(bs,
+                                                             ts,
+                                                             node_imp_list_ptr,
+                                                             class_count,
+                                                             node_id,
+                                                             imp_threshold,
+                                                             min_obs_leaf);
+                                }
+                                else {
+                                    sp_hlp.calc_imp_dec(ts, node_ptr, node_imp_list_ptr, node_id);
+                                    sp_hlp.choose_best_split(bs,
+                                                             ts,
+                                                             node_imp_list_ptr,
+                                                             impl_const_t::hist_prop_count_,
+                                                             node_id,
+                                                             imp_threshold,
+                                                             min_obs_leaf);
+                                }
+                            }
+
+                            ts.ftr_bin = sycl::reduce_over_group(item.get_group(),
+                                                                 bin > ts.ftr_bin ? bin : index_max,
+                                                                 minimum<Index>());
+                        }
+                    }
+
+                    if (sub_group_id > 0) {
```

Comment:
Is everything above is needs for sub_group_id > 0?

Reply:
Yes, other WI participate in wg cumulative operations for processing rows values for current node